ui/lib/pki/addon/components/key-usage.ts

```diff
@@ -0,0 +1,91 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+/**
+ * @module PkiKeyUsage
+ * PkiKeyUsage components are used to build out the toggle options for PKI's role create/update key_usage, ext_key_usage and ext_key_usage_oids model params.
+ * Instead of having the user search on the following goLang pages for these options we present them in checkbox form and manually add them to the params as an array of strings.
+ * key_usage options: https://pkg.go.dev/crypto/x509#KeyUsage
+ * ext_key_usage options (not all are include on purpose): https://pkg.go.dev/crypto/x509#ExtKeyUsage
+ * @example
+ * ```js
+ * <PkiKeyUsage @model={@model} @group={group}/>
+ * ```
+ * @param {class} model - The pki/pki-role-engine model.
+ * @param {string} group - The name of the group created in the model. In this case, it's the "Key usage" group.
+ */
+
+interface Field {
+  key: string;
+  label: string;
+}
+
+const KEY_USAGE_FIELDS: Field[] = [
+  { key: 'DigitalSignature', label: 'Digital Signature' },
+  { key: 'ContentCommitment', label: 'Content Commitment' },
+  { key: 'CrlSign', label: 'CRL Sign' },
+  { key: 'KeyAgreement', label: 'Key Agreement' },
+  { key: 'DataEncipherment', label: 'Data Encipherment' },
+  { key: 'EncipherOnly', label: 'Encipher Only' },
+  { key: 'KeyEncipherment', label: 'Key Encipherment' },
+  { key: 'CertSign', label: 'Cert Sign' },
+  { key: 'DecipherOnly', label: 'Decipher Only' },
+];
+
+const EXT_KEY_USAGE_FIELDS: Field[] = [
+  { key: 'Any', label: 'Any' },
+  { key: 'EmailProtection', label: 'Email Protection' },
+  { key: 'TimeStamping', label: 'Time Stamping' },
+  { key: 'ServerAuth', label: 'Server Auth' },
+  { key: 'IpsecEndSystem', label: 'IPSEC End System' },
+  { key: 'OcspSigning', label: 'OCSP Signing' },
+  { key: 'ClientAuth', label: 'Client Auth' },
+  { key: 'IpsecTunnel', label: 'IPSEC Tunnel' },
+  { key: 'IpsecUser', label: 'IPSEC User' },
+  { key: 'CodeSigning', label: 'Code Signing' },
+];
+
+interface PkiKeyUsageArgs {
+  model: {
+    keyUsage: string[];
+    extKeyUsageOids: string[];
+    extKeyUsage: string[];
+  };
+}
+
+export default class PkiKeyUsage extends Component<PkiKeyUsageArgs> {
+  keyUsageFields: Field[] = [];
+  extKeyUsageFields: Field[] = [];
+
+  constructor(owner: any, args: PkiKeyUsageArgs) {
+    super(owner, args);
+    this.keyUsageFields = KEY_USAGE_FIELDS;
+    this.extKeyUsageFields = EXT_KEY_USAGE_FIELDS;
+    // TODO: set this in the model by default
+    // set default of key_usage to the three params that are true by default.
+    this.args.model.keyUsage = ['DigitalSignature', 'KeyAgreement', 'KeyEncipherment'];
```

Comment:
perhaps outside the scope of this work, but I had commented on the original PR that this would overwrite the model's attrs when editing a role, right? 

I think it can be resolved by just defining the attr in `models/pki/role.js` and removing lines 66 and 67 all together.

```js
  @attr('array', {
    defaultValue() {
      return ['DigitalSignature', 'KeyAgreement', 'KeyEncipherment'];
    },
  })
  keyUsage;
```

This also addresses the type inconsistency with `keyUsage`. The [docs](https://developer.hashicorp.com/vault/api-docs/secret/pki#parameters-5) show the default value is an array of strings, but when console logging `this.args.model` or inspecting the ember dev console openApi sets it as a string list: 

<img width="583" alt="Screen Shot 2022-11-11 at 4 05 44 PM" src="https://user-images.githubusercontent.com/68122737/201446213-82cff4b2-9259-4235-b5af-68139f5f3956.png">

Reply:
Yes! My comment `// TODO: set this in the model by default` was about moving this to the model in the future, but it's not that big of a change so I can do that in this PR